system-contract-testing/utils/contract.js

```diff
@@ -1,11 +1,7 @@
 const { ethers } = require("hardhat");
 const { ONE_HBAR } = require("./constants");
 
-async function contractDeployAndFund(name, gas = 0, value = 0) {
-  // provider configs override
-  if (gas > 0) {
-    ethers.provider.estimateGas = async () => gas;
-  }
+async function contractDeployAndFund(name, value = 0) {
```

Comment:
What is the value of this indirection, vs using directly ethers?
On the other hand why using factory instead

ethers.deployContract() or deployer.deployContract()